### PR TITLE
Make pieman_devices package out of devices directory

### PR DESCRIPTION
**Pull request by eugulixes**
# Please make sure all below checkboxes in the Mandatory section have been checked before submitting your PR (also don't forget to pay attention to the Depending on Circumstances section)

## Mandatory

- [x] The commit message is an imperative and starts with a capital letter (for example, `Add something`, `Remove something`, `Fix something`, etc.).
- [x] I made sure I hadn't put a period at the end of the the commit message(s).

## Depending on Circumstances

Some of the items in the section become mandatory if you are a first-time contributor and/or your changes are relevant to the items.

- [ ] Adding a new parameter I didn't forget to reflect it in `README.md`.
- [ ] Adding a new parameter or modifying an existing one I didn't break the alphabetical order.
- [ ] Adding a new utility written in Python I didn't forget to reflect it in `pieman/README.md`.
- [ ] Adding a new utility written in Python I didn't forget to increase the version number of the [pieman](https://pypi.org/project/pieman/) package in `pieman/setup.py`, `pieman/pieman/__init__.py` and `essentials.sh`.
- [ ] (for first-time contributors) One of the commits in the pull request adds me to the **Acknowledgements** section in `AUTHORS.md`.

### CODE REVIEW

File: devices/bin/__init__.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+""""""
```
Comment (karpa4o4) [inline devices/bin/__init__.py:16]: @eugulixes, can you suggest a name for the module?

File: devices/setup.py [hunk 2]
```diff
@@ -105,8 +105,10 @@ def main():
         if not Path(path_to_dev).is_dir():
             continue  # skip setup.py and other regular files
 
-        if dev_name in ['build', 'dist', f'{PACKAGE_NAME}.egg-info', PACKAGE_NAME]:
-            continue  # skip the dirs produced as a result of 'setup.py sdist' or 'setup.py build'
+        if dev_name in ['bin', 'build', 'dist', f'{PACKAGE_NAME}.egg-info', PACKAGE_NAME]:
+            continue
+            # skip the dirs produced as a result of 'setup.py sdist' or 'setup.py build'
+            # and bin dir with source of pieman_devices package
```
Comment (eugulixes) [inline devices/setup.py:111]: `and the bin directory with the sources of the pieman_devices package.`

File: devices/bin/__init__.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+def _get_abs_path(rel_path):
+    """Returns the full path which is the result of concatenation of BASE_DIR and the specified
+    relative path.
+    """
+
+    return os.path.join(BASE_DIR, rel_path)
+
+
+def get_devices():
+    """Returns the list of devices supported in Pieman. """
```
Comment (eugulixes) [inline devices/bin/__init__.py:35]: `"""Returns the list of the devices supported by Pieman. """`

File: devices/bin/__init__.py (hunk 1 above)
Comment (eugulixes) [reply devices/bin/__init__.py:16]: `Module containing the interface for serving the devices supported by Pieman to clients.`

**Comment by eugulixes** [comment]
Put the following line to README.md as a good start: `The directory contains a tree-like representation of the devices and operating systems supported by Pieman.`

File: devices/__init__.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Module containing the interface for serving the devices supported by Pieman to clients. """
```
Comment (eugulixes) [inline devices/__init__.py:16]: ...devices and operating systems...

**Review by karpa4o4 — APPROVED** [review]

**Pull request merged**